src/d3a/models/strategy/external_strategies/load.py

```diff
@@ -127,15 +134,17 @@ def _bid(self, payload):
             return
         try:
             arguments = json.loads(payload["data"])
-            assert set(arguments.keys()) == {'price', 'energy', 'transaction_id'}
+            assert all(arg in self.ALLOWED_PARAMETERS for arg in arguments.keys())
```

Comment:
This check allows us to have optional arguments (like `replace_existing`). However, this is semantically different than the previous check:
```python
assert set(arguments.keys()) == {'price', 'energy', 'transaction_id'}
```
Because now these 3 arguments are not mandatory anymore.

Let me know if this needs to be changed: in that case, I will create a separate function or class to validate the input arguments for the bid/offer commands.

Reply:
IMO you dont need this, as `_get_transaction_id` (line 131) will already fail if there is no `transaction_id` in the payload.

Reply:
And as this is done in every callback function, there is no need for `ALLOWED_PARAMETERS` and `ALLOWED_PARAMETERS` , as these are only applicable for commands like `bid` or `offer` . E.g. for _delete_offer, we need the `offer_id` as input parameter

Reply:
Ok, since the `ALLOWED_PARAMETERS` would be different depending on the method (bid/offer/delete etc) it doesn't make that much sense to have them as a class constant. I suppose it would be nice to have a separate validator class that defines these parameters for each method, maybe we can think of it for a different task!

I'll then replace the `ALLOWED_PARAMETERS` with a standard set defined in the function itself, for the moment :)